java/yb-cql/src/test/java/org/yb/cql/CQLTester.java

```diff
@@ -28,180 +28,267 @@
 import org.yb.client.TestUtils;
 
 import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.LinkedHashMap;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableSet;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 /**
- * Base class for CQL tests.
- */
+* Base class for CQL tests.
+*/
 public class CQLTester extends BaseCQLTest
 {
-    public static final String KEYSPACE = "cql_test_keyspace";
-    private static final AtomicInteger seqNumber = new AtomicInteger();
+  public static final String KEYSPACE = "cql_test_keyspace";
+  private static final AtomicInteger seqNumber = new AtomicInteger();
 
-    private List<String> tables = new ArrayList<>();
+  private List<String> tables = new ArrayList<>();
+  private List<String> indexes = new ArrayList<>();
 
-    protected void createTable(String query)
-    {
-        createTable(KEYSPACE, query);
-    }
+  protected void createTable(String query)
+  {
+    createTable(KEYSPACE, query);
+  }
 
-    protected void createTable(String keyspace, String query)
-    {
-        String currentTable = createTableName();
-        String fullQuery = formatQuery(keyspace, query);
-        final long startTimeMs = System.currentTimeMillis();
-        session.execute(fullQuery);
-        long elapsedTimeMs = System.currentTimeMillis() - startTimeMs;
-        LOG.info("After execute: {} took {}ms", fullQuery, elapsedTimeMs);
-    }
+  protected void createIndex(String query)
+  {
+    createIndex(KEYSPACE, query);
+  }
 
-    protected String createTableName()
-    {
-        String currentTable = "table_" + seqNumber.getAndIncrement();
-        tables.add(currentTable);
-        return currentTable;
-    }
+  protected void createTable(String keyspace, String query)
+  {
+    String currentTable = createTableName();
+    String fullQuery = formatQuery(keyspace, query);
+    final long startTimeMs = System.currentTimeMillis();
+    session.execute(fullQuery);
+    long elapsedTimeMs = System.currentTimeMillis() - startTimeMs;
+    LOG.info("After createTable execute: {} took {}ms", fullQuery, elapsedTimeMs);
+  }
 
-    protected String formatQuery(String query)
-    {
-        return formatQuery(KEYSPACE, query);
-    }
+  protected void createIndex(String keyspace, String query)
+  {
+    String fullQuery = String.format(query, keyspace + "_" + createIndexName(), keyspace + "." + currentTable());
```

Comment:
As explained separately, index name should be a simple name without keyspace. Please remove `keyspace` and `_` from the index name.